cohort-evaluator/src/main/java/com/ibm/cohort/cql/evaluation/CqlContextFactory.java

```diff
@@ -125,14 +133,14 @@ public Context createContext(CqlLibraryProvider libraryProvider, CqlLibraryDescr
             CqlTerminologyProvider terminologyProvider, CqlDataProvider dataProvider, ZonedDateTime evaluationDateTime,
             Pair<String, String> contextData, Map<String, Parameter> parameters, CqlDebug debug)
             throws CqlLibraryDeserializationException {
-        
+
         ContextCacheKey key = new ContextCacheKey( libraryProvider, topLevelLibrary, terminologyProvider, evaluationDateTime, parameters );
```

Comment:
Did you consider whether or not the external function provider needs to be part of the context cache key? The current implementation is stateless, so it certainly isn't going to matter for the Spark application, but the CqlContextFactory lives in cql-evaluator which is intended to be the new general use API down the line. It feels to me like it is most correct to include the external function provider in the cache key.